lib/__tests__/Button.test.js

```diff
@@ -52,6 +52,21 @@ describe('<Button>', function() {
     expect(shallowToJson(wrapper)).toMatchSnapshot();
   });
 
+  it('DOM with some props and component instance used instead of tag', function() {
+    var button = el(
+      Button,
+      {
+        id: 'foo',
+        className: 'bar',
+        tag: 'div',
+        component: el('button'),
```

Comment:
I would prefer that the API does not involve passing an Element in, but the Component itself. That is, instead of passing `<MySpecialComponent />` you pass `MySpecialComponent`. Does that make sense?